codespell_lib/_codespell.py

```diff
@@ -983,6 +984,20 @@ def parse_file(
     return bad_count
 
 
+def flatten_comma_separated_arguments(
+    arguments: List[str],
+) -> List[str]:
+    """
+    >>> flatten_comma_separated_arguments(["a,b,c", "d", "e"])
+    ['a', 'b', 'c', 'd', 'e']
+    >>> flatten_comma_separated_arguments([])
+    []
+    >>> flatten_comma_separated_arguments(None)
+    []
+    """
+    return [item for argument in arguments or [] for item in argument.split(",")]
```

Comment:
Arguments is always list of str so why do you need the `arguments or []`?
```suggestion
    return [item for argument in arguments for item in argument.split(",")]
```

Reply:
Because the doctest on lines 995 and 996 will fail.

Reply:
I was trying not to increase complexity by avoiding `if` where this function is called.

I had ended up adding an `or []` in the function arguments, then in the function itself to avoid multiple `or []` wherever it is called (and for leniency).

And now I am back to `if` because that's what's makes sense IMHO: do not call the function if that's not needed, and for that, you do need the `if`'s.

Reply:
In all case, I was actually battling against complexity metrics that don't want new `if`s, and perceived complexity too.

First added `or []` in function arguments, wherever we call the function.

Then decided it might be better to have a single location for `or []`, at the expense of changing the signature of the function to accept a `None` argument too. But since mypy doesn't catch such violations, I thought that might be a feature and not a mypy shortcoming. So, I left the signature as is – to avoid increasing perceived complexity even more.

Then I went back to the first solution, but kept the second solution for leniency. But I agree that's over-engineering.

And finally, I'm back to the original code. And I think we do need to **accept increased complexity** when slightly modifying existing options or adding new options. The only right way to reduce complexity here is to modify the way we handle **all** options, **consistently**.